Support custom capture resolution in CamCapture

tvdstaaij opened the pull request:
This patch allows you to override the camera resolution to something other than the default (cameras often support multiple capture resolutions) like this:

```
import lcvcore 1.0

Rectangle{
    CamCapture{
        device: '0'
        width: 1280
        height: 720
    }
}
```

Not sure whether I did this right, played around a bit with different approaches and this seems to work well for me. I added the `reinitializeDevice` construction after I found that without starting a new capture there would sometimes be errors about the camera device being busy, effectively crashing the capture (probably some kind of race condition?).

dinusv commented:
Hi Tim,

Thanks for the request, just a small change: the name of the ```width``` and ```height``` properties since they are already inherited from ```QQuickItem```:```QCamCapture``` < ```QMatDisplay``` < ```QQuickItem```. They control the display size of the mat. I would add them under a single property of ```QSize``` type:

```
CamCapture{
   device: '0'
   resolution: Qt.size(1280,720)
 }
```

Having them under a single property also helps synchronize both values given by the user in live coding, where the code may recompile before the user has added both the width and height properties, hence sending an incompatible resolution to the camera.

If you could make this small change I would be happy to merge,
Thanks!

tvdstaaij commented:
I was already in doubt whether to implement it as two primitive type properties or one class type property, but hadn't realized it was also shadowing parent properties. I didn't think of using QSize, looks clean and integrates nicely with Qt Quick. So yeah,this is definitely the way to go.

I updated my commit with the QSize refactor and a few other improvements I thought of. I also updated the qdoc comments to include the new property and demonstrate its usage.

Perhaps interesting to know, OpenCV seems to fall back to the next lowest (or closest? not sure) supported resolution when an unsupported resolution is set (e.g. 1300x800 -> 1280x720 if the camera supports 720p). I added a qWarning trigger whenever this happens, so the user can see that OpenCV replaced their intended resolution with a fallback resolution.

dinusv commented:
Perfect!

I'm not sure of the behavior for unsupported resolutions, it might even depend on the camera, but the warning there's good.
